services/dis-apim-operator/api/v1alpha1/apiversion_type_getazurediag_test.go

```diff
@@ -0,0 +1,100 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/Altinn/altinn-platform/services/dis-apim-operator/internal/utils"
+	apim "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/apimanagement/armapimanagement/v2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestGetAzureAPIAppInsightsDiagnosticSettings(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ApiVersion Suite")
+}
+
+var _ = Describe("ApiVersion", func() {
+	var (
+		apiVersion      ApiVersion
+		defaultLoggerID string
+	)
+	BeforeEach(func() {
+		defaultLoggerID = "defaultLogger"
+		apiVersion = ApiVersion{
+			Spec: ApiVersionSpec{
+				ApiVersionSubSpec: ApiVersionSubSpec{
+					Diagnostics: &ApiDiagnosticSpec{
+						LoggerName:         &defaultLoggerID,
+						SamplingPercentage: utils.ToPointer(int32(75)),
+						EnableMetrics:      utils.ToPointer(false),
+						Frontend: &PipelineDiagnosticSettings{
+							Request: &HttpMessageDiagnostic{
+								Headers: []*string{utils.ToPointer("Frontend-Request-Header")},
+							},
+							Response: &HttpMessageDiagnostic{
+								Headers: []*string{utils.ToPointer("Frontend-Response-Header")},
+							},
+						},
+						Backend: &PipelineDiagnosticSettings{
+							Request: &HttpMessageDiagnostic{
+								Headers: []*string{utils.ToPointer("Backend-Request-Header")},
+							},
+							Response: &HttpMessageDiagnostic{
+								Headers: []*string{utils.ToPointer("Backend-Response-Header")},
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+
+	Describe("GetAzureAPIDiagnosticSettings", func() {
+		It("should return the specified diagnostic settings for the api", func() {
+			diagnosticSettings := apiVersion.GetAzureAPIAppInsightsDiagnosticSettings(defaultLoggerID)
+
+			Expect(diagnosticSettings.Properties.LoggerID).To(Equal(&defaultLoggerID))
+			Expect(diagnosticSettings.Properties.Sampling.Percentage).To(Equal(utils.ToPointer(75.0)))
```

Comment:
Small nitpick: In the [setup](https://github.com/Altinn/altinn-platform/pull/1473/files#diff-aee6e0b74698a5c1680d1bca32cc7d7409f878f8b1fa5aba2b516b9af3fd1a4aR29) you have used `utils.ToPointer(int32(75))`.

Would it be better to make these as variables instead of duplicating the values?
It will make writing future tests easier as well: [TableDrivenTests](https://go.dev/wiki/TableDrivenTests), [Ginkgo](https://onsi.github.io/ginkgo/#table-specs) relevant docs.

Reply:
CRD defines sample rate as int, azure api uses float. I didn't feel like we needed expose the need for defining sampling percentage with decimals. Thats the reason for them being different. 
I dont see the need for future tests changing this as overriding and default value of the field is already tested.